internal/lldp/rx.go

```diff
@@ -335,52 +357,54 @@ func (l *LLDP) stopCapture() error {
 	l.mu.Lock()
 	defer l.mu.Unlock()
 
-	if !l.rxRunning {
+	if !l.Rx.Running {
 		return nil // Already stopped
 	}
 
-	logger := l.l.With().Str("interface", l.interfaceName).Logger()
+	logger := l.Rx.Logger
+	rxCancel := l.Rx.Cancel
+	tPacket := l.Rx.TPacket
+
 	logger.Info().Msg("stopping LLDP receiver")
 
 	// Cancel context to signal stop
-	rxCancel := l.rxCancel
 	if rxCancel != nil {
+		l.Rx.Cancel = nil // so we don't cancel again
 		rxCancel()
-		l.rxCancel = nil
-
 		logger.Info().Msg("cancelled RX context, waiting for goroutine to finish")
 	}
 
 	// stop the TPacketRx
 	go func() {
-		if l.tPacketRx == nil {
+		if tPacket == nil {
 			return
 		}
 
 		// write an empty packet to the TPacketRx to interrupt the blocking read
 		// it's a shitty workaround until https://github.com/google/gopacket/pull/777 is merged,
 		// or we have a better solution, see https://github.com/google/gopacket/issues/1064
-		_ = l.tPacketRx.WritePacketData([]byte{})
+		_ = tPacket.WritePacketData([]byte{})
 	}()
 
 	// wait for the goroutine to finish
 	start := time.Now()
 	l.rxWaitGroup.Wait()
 	logger.Info().Dur("duration", time.Since(start)).Msg("RX goroutine finished")
 
-	l.rxRunning = false
-
-	if l.tPacketRx != nil {
-		logger.Info().Msg("closing TPacketRx")
-		l.tPacketRx.Close()
-		l.tPacketRx = nil
-	}
+	l.Rx.Running = false
```

Comment:
Eagerly clear the running flag because we're about to tear everything down in reverse order.